pilot/cmd/pilot-agent/status/server.go

```diff
@@ -243,30 +267,58 @@ type PrometheusScrapeConfiguration struct {
 // the application metrics and merge them together.
 // The merge here is a simple string concatenation. This works for almost all cases, assuming the application
 // is not exposing the same metrics as Envoy.
-// TODO(https://github.com/istio/istio/issues/22825) expose istio-agent stats here as well
+// Note that we do not return any errors here. If we do, we will drop metrics. For example, the app may be having issues,
+// but we still want Envoy metrics. Instead, errors are tracked in the failed scrape metrics/logs.
 func (s *Server) handleStats(w http.ResponseWriter, r *http.Request) {
-	var envoy, application []byte
+	scrapeTotals.Increment()
+	var envoy, application, agent []byte
 	var err error
+	// Gather all the metrics we will merge
 	if envoy, err = s.scrape(fmt.Sprintf("http://localhost:%d/stats/prometheus", s.envoyStatsPort), r.Header); err != nil {
-		log.Errora(err)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
+		log.Errorf("failed scraping envoy metrics: %v", err)
+		envoyScrapeErrors.Increment()
 	}
 	if s.prometheus != nil {
 		url := fmt.Sprintf("http://localhost:%s%s", s.prometheus.Port, s.prometheus.Path)
 		if application, err = s.scrape(url, r.Header); err != nil {
-			log.Errora(err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
+			log.Errorf("failed scraping application metrics: %v", err)
+			appScrapeErrors.Increment()
 		}
 	}
+	if agent, err = scrapeAgentMetrics(); err != nil {
+		log.Errorf("failed scraping agent metrics: %v", err)
+		agentScrapeErrors.Increment()
+	}
+
+	// Write out the metrics
 	if _, err := w.Write(envoy); err != nil {
 		log.Errorf("failed to write envoy metrics: %v", err)
-		return
+		envoyScrapeErrors.Increment()
 	}
 	if _, err := w.Write(application); err != nil {
 		log.Errorf("failed to write application metrics: %v", err)
+		appScrapeErrors.Increment()
+	}
+	if _, err := w.Write(agent); err != nil {
+		log.Errorf("failed to write agent metrics: %v", err)
+		appScrapeErrors.Increment()
+	}
+}
+
+func scrapeAgentMetrics() ([]byte, error) {
+	buf := &bytes.Buffer{}
+	mfs, err := promRegistry.Gather()
+	enc := expfmt.NewEncoder(buf, expfmt.FmtText)
```

Comment:
From **https://godoc.org/github.com/prometheus/common/expfmt#NewEncoder** there should also be a call `defer enc.Close()` statement. It doesn't look like the interface implements Close although the documentation refers to it.

Reply:
Pulled down the branch. No such method exists. LGTM.